iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -265,6 +273,28 @@ public String getFileFormatPropertyKey() {
     return TableProperties.DEFAULT_FILE_FORMAT;
   }
 
+  @Override
+  public boolean useNativeCommit() {
+    return true;
+  }
+
+  @Override
+  public void nativeCommit(Properties commitProperties, boolean overwrite) throws HiveException {
+    String tableName = commitProperties.getProperty(Catalogs.NAME);
+    Configuration configuration = SessionState.getSessionConf();
+    Optional<JobContext> jobContext = getJobContextForCommitOrAbort(configuration, tableName, overwrite);
+    if (jobContext.isPresent()) {
+      try {
+        OutputCommitter committer = new HiveIcebergOutputCommitter();
+        committer.commitJob(jobContext.get());
+      } catch (Throwable e) {
+        LOG.error("Error while trying to commit job, starting rollback", e);
+        rollbackInsertTable(configuration, tableName, overwrite);
+        throw new HiveException("Error committing job", e);
```

Comment:
Might be worth including the jobID and tableName into the exception message here

Reply:
Makes sense 😄

Reply:
Done